### PR TITLE
Add HTMLReader, sample app and HTML file

### PR DESCRIPTION
**Pull request by mtutty**
Pretty much what the title says.  Adds HTMLReader to core/src/readers, a sample ts file in apps/simple, and a pretty gnarly HTML file in apps/simple/data.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -4,5 +4,11 @@
   "editor.defaultFormatter": "esbenp.prettier-vscode",
   "[xml]": {
     "editor.defaultFormatter": "redhat.vscode-xml"
+  },
```
Comment (yisding) [inline .vscode/settings.json:7]: Can you revert the changes here? We want to use prettier if possible.

Comment (mtutty) [reply .vscode/settings.json]: Yup, check it now.  That wasn't a conscious choice, but suggested by vscode while I was working in the IDE.  If you're doing linting stuff on commit/push, maybe the .vscode dir should be ignored?

**Review by yisding — APPROVED** [review]
OK this look good. Thank you for doing this!
